npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser.rb

```diff
@@ -334,6 +340,10 @@ def url_for_relevant_cred(resolved_url)
         resolved_url.gsub(/#{Regexp.quote(reg)}.*/, "") + reg
       end
 
+      def submodule_package_files
+        @submodule_package_files ||= package_files.select { |f| f.type == "submodule" }
```

Comment:
is there any way to have test coverage? So we can catch regressions in the future easily

Reply:
I think a smoke test would be best, I can work on it!

Reply:
A smoke test turned out to be tricky, so I added one regular spec to cover this, and corrected a bug it found.